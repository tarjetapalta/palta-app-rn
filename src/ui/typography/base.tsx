import React, { FunctionComponent } from 'react';
import { ViewProps, TextStyle } from 'react-native';

import { KeyColors } from '@theme/colors';
import { KeyEmphasis } from '@theme/emphasis';
import styled from '@theme/styled-components';
import { KeyFontSize, KeyFontWeight } from '@theme/text';

type FontsType =
  | 'BwModelica-Black'
  | 'BwModelica-BlackItalic'
  | 'BwModelica-Bold'
  | 'BwModelica-BoldItalic'
  | 'BwModelica-ExtraBold'
  | 'BwModelica-ExtraBoldItalic'
  | 'BwModelica-Hairline'
  | 'BwModelica-HairlineItalic'
  | 'BwModelica-Light'
  | 'BwModelica-LightItalic'
  | 'BwModelica-Medium'
  | 'BwModelica-MediumItalic'
  | 'BwModelica-Regular'
  | 'BwModelica-RegularItalic'
  | 'BwModelica-Thin'
  | 'BwModelica-ThinItalic';

type TextType = Pick<ViewProps, 'testID'>;

export interface TextProps extends TextType, Pick<TextStyle, 'textAlign'> {
  color?: KeyColors;
  size?: KeyFontSize;
  weight?: KeyFontWeight;
  emphasis?: KeyEmphasis;
  numberOfLines?: number;
}

const setWeight = (weight: KeyFontWeight): FontsType =>
  `BwModelica-${weight}` as FontsType;

export const TextDefault = styled.Text<TextProps>`
  color: ${({ theme, color }) =>
    color ? theme.colors[color] : theme.colors.text};
  font-size: ${({ theme, size }) =>
    size ? theme.text.fontSize[size] : theme.text.fontSize.m};
  opacity: ${({ theme, emphasis }) =>
    emphasis ? theme.emphasis[emphasis] : theme.emphasis.high};
  text-align: ${({ textAlign }) => textAlign || 'center'};
  font-family: ${({ theme, weight }) =>
    weight
      ? setWeight(theme.text.fontWeight[weight] as KeyFontWeight)
      : 'BwModelica-Regular'};
`;

export const Text: FunctionComponent<TextProps> = ({
  /** Test id for test */
  testID,
  /** Text */
  children,
  /** Text color */
  color,
  /** Font size */
  size,
  /** Text font weight */
  weight,
  /** Text font opacity */
  emphasis,
  /** Text align left center or right */
  textAlign,
  /** Number of line to truncate text */
  numberOfLines,
}) => (
  <TextDefault
    testID={testID}
    color={color}
    size={size}
    weight={weight}
    emphasis={emphasis}
    textAlign={textAlign}
    numberOfLines={numberOfLines || 0}
  >
    {children}
  </TextDefault>
);
