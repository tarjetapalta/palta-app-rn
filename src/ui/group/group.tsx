import React, { FunctionComponent } from 'react';
import { ViewProps } from 'react-native';

import styled from '@theme/styled-components';

type KeyGroup =
  | 'header'
  | 'headerModal'
  | 'body'
  | 'bodyWithButtonContent'
  | 'bodyWithButton'
  | 'footer'
  | 'borderless';
type KeyContent = 'left' | 'center' | 'right';

type GroupType = Pick<ViewProps, 'testID'>;

interface GroupProps extends GroupType {
  type?: KeyGroup;
  content?: KeyContent;
}

const WrapperGroup = styled.View<Pick<GroupProps, 'type' | 'content'>>`
  padding: 0 ${({ theme }) => theme.spacers.l};
  ${({ theme, type }) =>
    type === 'header' &&
    `
    padding-top: ${theme.spacers.l};
  `};
  ${({ type, theme }) =>
    type === 'headerModal' &&
    `
    padding-top: ${theme.spacers.xs};
  `};
  ${({ type }) =>
    type === 'body' &&
    `
    flex: 1;
  `};
  ${({ theme, type }) =>
    type === 'bodyWithButtonContent' &&
    `
    padding: 0 ${theme.spacers.l};
  `};
  ${({ theme, type }) =>
    type === 'bodyWithButton' &&
    `
    flex: 1;
    padding: 0 ${theme.spacers.xs};
  `};
  ${({ theme, type }) =>
    type === 'footer' &&
    `
    padding: 0 ${theme.spacers.xs};
    padding-bottom: ${theme.spacers.l};
  `};
  ${({ type }) =>
    type === 'borderless' &&
    `
    padding: 0px;
  `};
  ${({ content }) =>
    content === 'left' &&
    `
    align-items: flex-start;
  `}
  ${({ content }) =>
    content === 'center' &&
    `
    justify-content: center;
    align-items: center;
  `}
  ${({ content }) =>
    content === 'right' &&
    `
    align-items: flex-end;
  `}
`;

export const Group: FunctionComponent<GroupProps> = ({
  /** Test id for test */
  testID,
  /** Pass children components */
  children,
  /** Specific type depending on the area it occupies on the screen, header | body | footer */
  type,
  /** Makes the content, left | right, for default content is center  */
  content = 'center',
}) => (
  <WrapperGroup testID={testID} type={type} content={content}>
    {children}
  </WrapperGroup>
);
