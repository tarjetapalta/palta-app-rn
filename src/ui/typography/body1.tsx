import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const Body1: FunctionComponent<
  Pick<
    TextProps,
    'testID' | 'color' | 'emphasis' | 'textAlign' | 'numberOfLines'
  >
> = ({
  /** Test id for test */
  testID = 'Body1',
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
  /** Number of line to truncate text */
  numberOfLines,
}) => (
  <Text
    testID={testID}
    color={color}
    size="s"
    weight="light"
    emphasis={emphasis}
    textAlign={textAlign}
    numberOfLines={numberOfLines}
  >
    {children}
  </Text>
);
