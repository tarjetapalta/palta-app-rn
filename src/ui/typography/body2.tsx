import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const Body2: FunctionComponent<
  Pick<TextProps, 'testID' | 'color' | 'emphasis' | 'textAlign'>
> = ({
  /** Test id for test */
  testID,
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="xs"
    weight="regular"
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
