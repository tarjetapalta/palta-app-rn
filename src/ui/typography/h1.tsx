import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const H1: FunctionComponent<TextProps> = ({
  /** Test id for test */
  testID = 'H1',
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="xxl"
    weight="bold"
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
