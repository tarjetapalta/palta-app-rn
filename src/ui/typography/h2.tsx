import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const H2: FunctionComponent<
  Pick<TextProps, 'testID' | 'color' | 'emphasis' | 'textAlign'>
> = ({
  /** Test id for test */
  testID = 'H2',
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="xl"
    weight="bold"
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
