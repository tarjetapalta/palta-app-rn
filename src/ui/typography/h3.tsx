import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const H3: FunctionComponent<TextProps> = ({
  /** Test id for test */
  testID = 'H3',
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="l"
    weight="medium"
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
