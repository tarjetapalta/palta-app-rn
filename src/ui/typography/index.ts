export { H1 } from './h1';
export { H2 } from './h2';
export { H3 } from './h3';
export { H4 } from './h4';
export { H5 } from './h5';
export { Body1 } from './body1';
export { Body2 } from './body2';
export { Overline } from './overline';
