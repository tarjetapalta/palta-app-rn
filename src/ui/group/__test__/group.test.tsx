import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { H1 } from '../../typography';
import { Group } from '../group';

describe('group component', () => {
  it('basic Group with type "body", content "center" and testId "Group"', () => {
    const result = renderer.create(
      <WithTheme>
        <Group>
          <H1>Group test</H1>
        </Group>
      </WithTheme>,
    );

    expect(result).toMatchSnapshot();
  });
  it('basic Group with type "header"', () => {
    const result = renderer.create(
      <WithTheme>
        <Group type="header">
          <H1>Group test</H1>
        </Group>
      </WithTheme>,
    );

    expect(result).toMatchSnapshot();
  });
  it('basic Group with content "left"', () => {
    const result = renderer.create(
      <WithTheme>
        <Group content="left">
          <H1>Group test</H1>
        </Group>
      </WithTheme>,
    );

    expect(result).toMatchSnapshot();
  });
  it('basic Group with custom testID', () => {
    const result = renderer.create(
      <WithTheme>
        <Group testID="custom testID">
          <H1>Group test</H1>
        </Group>
      </WithTheme>,
    );

    expect(result).toMatchSnapshot();
  });
});
