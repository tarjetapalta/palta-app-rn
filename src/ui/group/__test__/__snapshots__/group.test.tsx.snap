// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group component basic Group with content "left" 1`] = `
<View
  content="left"
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "paddingBottom": 0,
        "paddingLeft": 48,
        "paddingRight": 48,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    numberOfLines={0}
    size="xxl"
    style={
      Array [
        Object {
          "color": "#283238",
          "fontFamily": "BwModelica-Bold",
          "fontSize": 28,
          "opacity": 0.87,
          "textAlign": "center",
        },
      ]
    }
    testID="H1"
    weight="bold"
  >
    Group test
  </Text>
</View>
`;

exports[`group component basic Group with custom testID 1`] = `
<View
  content="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 48,
        "paddingRight": 48,
        "paddingTop": 0,
      },
    ]
  }
  testID="custom testID"
>
  <Text
    numberOfLines={0}
    size="xxl"
    style={
      Array [
        Object {
          "color": "#283238",
          "fontFamily": "BwModelica-Bold",
          "fontSize": 28,
          "opacity": 0.87,
          "textAlign": "center",
        },
      ]
    }
    testID="H1"
    weight="bold"
  >
    Group test
  </Text>
</View>
`;

exports[`group component basic Group with type "body", content "center" and testId "Group" 1`] = `
<View
  content="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 48,
        "paddingRight": 48,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    numberOfLines={0}
    size="xxl"
    style={
      Array [
        Object {
          "color": "#283238",
          "fontFamily": "BwModelica-Bold",
          "fontSize": 28,
          "opacity": 0.87,
          "textAlign": "center",
        },
      ]
    }
    testID="H1"
    weight="bold"
  >
    Group test
  </Text>
</View>
`;

exports[`group component basic Group with type "header" 1`] = `
<View
  content="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 48,
        "paddingRight": 48,
        "paddingTop": 48,
      },
    ]
  }
  type="header"
>
  <Text
    numberOfLines={0}
    size="xxl"
    style={
      Array [
        Object {
          "color": "#283238",
          "fontFamily": "BwModelica-Bold",
          "fontSize": 28,
          "opacity": 0.87,
          "textAlign": "center",
        },
      ]
    }
    testID="H1"
    weight="bold"
  >
    Group test
  </Text>
</View>
`;
